server/src/controllers/grades.ts

```diff
@@ -4,27 +4,139 @@
 
 import { parse, Parser } from 'csv-parse';
 import { NextFunction, Request, Response } from 'express';
+import { Op, Transaction } from 'sequelize';
 
+import { sequelize } from '../database';
+import Attainable from '../database/models/attainable';
+import CourseInstanceRole from '../database/models/courseInstanceRole';
+import User from '../database/models/user';
+import UserAttainmentGrade from '../database/models/userAttainmentGrade';
+
+import { CourseInstanceRoleType } from '../types/course';
 import { ApiError } from '../types/error';
 import { idSchema } from '../types/general';
+import { AttainmentGrade, Grade, Student } from '../types/grades';
 import { HttpCode } from '../types/httpCode';
 import { validateCourseAndInstance } from './utils/courseInstance';
 
+/**
+ * Parse and extract attainment id's from the CSV file header.
+ * Correct format: "StudentNo,C3I9A1,C3I9A2,C3I9A3,C3I9A4,C3I9A5..."
+ * @param {Array<string>} header - Header part of the CSV file.
+ * @returns {Array<number>} - Array containing the id's of attainments.
+ * @throws {ApiError} - If first column not "StudentNo" (case-insensitive)
```

Comment:
This uses "StudentNo" to match the requirements in frontend, I do agree that "StudentNumber" is more self-describing. I'll leave it as is for now.

Reply:
Or would it be better not to check the first column at all (it could be anything, e.g, "studentNo", "studentNumber", "-", etc.). We don't really use the "StudentNo" (row 1, col 1) for anything in the back end. This header 1 row 1 col check in the backend was more of a just to enforce strict style to the CSV teachers upload.